### PR TITLE
docs: Add Installation Warning and Kustomize Guide

### LINKED ISSUE #19872: docs: Add Installation Warning and Kustomize Guide
# Summary
The current ArgoCD installation guide suggests deploying components in the argocd namespace by default. However, the cluster-rbac ClusterRoleBinding specifies the ServiceAccount namespace as argocd, and changing this namespace can cause issues. It would be beneficial to update the documentation with a warning about this.

Additionally, I suggest adding instructions on how to modify the namespace during installation using kustomize.

# Motivation
We are running multiple ArgoCD instances within a single cluster for different purposes. However, we encountered errors when attempting to change the namespace because the ClusterRoleBinding specifies the ServiceAccount namespace as argocd. There is currently no warning about this issue in the documentation, which led to spending considerable time debugging.

Providing clear guidance on this issue and adding instructions on how to install ArgoCD in a custom namespace using kustomization would be helpful.

# Proposal
Update the ArgoCD installation guide to clearly warn users not to change the ServiceAccount namespace in the ClusterRoleBinding.
Add a section to the documentation with a guide on how to install ArgoCD in a different namespace using kustomize, including an example kustomization.yaml for reference.

### PR DESCRIPTION
**Pull request by nueavv**
closes: #19872 

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: docs/operator-manual/installation.md [hunk 1]
```diff
@@ -78,6 +81,30 @@ resources:
...
+  - https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.2/manifests/install.yaml
+
+patches:
+  - patch: |-
+      - op: replace
+        path: /subjects/0/namespace
+        value: <your-custom-namespace>
+    target:
+      kind: ClusterRoleBinding
+      name: <name-of-clusterrolebinding>
```
Comment (nitishfy) [inline docs/operator-manual/installation.md:103]:
```suggestion
      name: <name-of-clusterrolebinding>
```
shouldn't we mention the name of the clusterrolebinding here?

**Review by nitishfy — COMMENTED** [review]
PTAL, thanks!

**Review by wanghong230 — APPROVED** [review]
Makes sense

**Review by todaywasawesome — APPROVED** [review]
Good addition

**Pull request merged**